include/aikido/planner/ConfigurationToEndEffectorOffset.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef AIKIDO_PLANNER_PLANTOENDEFFECTOROFFSET_HPP_
+#define AIKIDO_PLANNER_PLANTOENDEFFECTOROFFSET_HPP_
+
+#include <dart/dart.hpp>
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/planner/Problem.hpp"
+#include "aikido/statespace/StateSpace.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Planning problem to plan to desired en-effector offset with fixed
+/// orientation
+class ConfigurationToEndEffectorOffset : public Problem
+{
+public:
+  /// Constructor.
+  ///
+  /// \param stateSpace State space.
+  /// \param bodyNode Body Node or robot for which the path is to be planned.
+  /// \param startState Start state.
+  /// \param goalState Goal state.
+  /// \param direction Direction of motion [unit vector in the world frame].
+  /// \param distance Distance to move, in meters.
+  /// \param interpolator Interpolator used to produce the output trajectory.
+  /// \param constraint Trajectory-wide constraint that must be satisfied.
+  /// \throw If \c stateSpace is not compatible to \c constraint's state space.
+  ConfigurationToEndEffectorOffset(
+      statespace::StateSpacePtr stateSpace,
+      dart::dynamics::BodyNodePtr bodyNode,
+      const statespace::StateSpace::State* startState,
+      const Eigen::Vector3d& direction,
+      const double& distance,
+      statespace::InterpolatorPtr interpolator,
+      constraint::TestablePtr constraint);
+
+  /// Returns the name of the planner problem.
+  const std::string& getName() const override;
+  static const std::string& getStaticName();
+
+  /// Returns the body node or robot for which the path is to be planned.
+  dart::dynamics::BodyNodePtr getBodyNode();
+
+  /// Returns the start state.
+  const statespace::StateSpace::State* getStartState() const;
+
```

Comment:
Like @brianhou mentioned, none of the problems except for `ConfigurationToConfiguration` have setters for start and goal.